model/riscv_jalr_seq.sail

```diff
@@ -9,26 +9,14 @@
 /* The definition for the sequential model. */
 
 function clause execute (JALR(imm, rs1, rd)) = {
-/* For the sequential model, the memory-model definition doesn't work directly
- * if rs1 = rd.  We would effectively have to keep a regfile for reads and another for
- * writes, and swap on instruction completion.  This could perhaps be optimized in
- * some manner, but for now, we just keep a reordered definition to improve simulator
- * performance.
- */
-  let t : xlenbits = X(rs1) + sign_extend(imm);
-  /* Extensions get the first checks on the prospective target address. */
-  match ext_control_check_addr(t) {
-    Ext_ControlAddr_Error(e) =>
-      Ext_ControlAddr_Check_Failure(e),
-    Ext_ControlAddr_OK(addr) => {
-      let target = [bits_of(addr) with 0 = bitzero];  /* clear addr[0] */
```

Comment:
This weirdo special behaviour for JALR has been lost

Reply:
I fixed JALR and added an assertion in `jump_to` to check the lowest bit is 0.

Arguably we should have `newtype pc_addr = PcAddr : bits(xlen - 1)` which omits that bit entirely... but that seemed like it might just be confusing and overkill. (But I would be open to being persuaded that we should; types are good.)